### PR TITLE
FOGL-2308 Fix reading of Python timestamp in different timezones

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/common/reading.cpp [hunk 1]
```diff
@@ -283,3 +267,70 @@ ostringstream assetTime;
...
+		int h, m;
+		int sign = (*ptr == '+' ? -1 : +1);
+		ptr++;
+		sscanf(ptr, "%02d:%02d", &h, &m);
+		ts->tv_sec += sign * ((3600 * h) + (60 * m));
+	}
+	// Now convert to the local time
+	extern long timezone;
+	tzset();
+	ts->tv_sec -= timezone;
```
Comment (AmandeepArora) [inline C/common/reading.cpp]: Will this be subtraction or addition? I think local time is "timezone" seconds ahead of UTC.

Comment (MarkRiddoch) [reply C/common/reading.cpp]: Yes, for timezones east of Greenwich timezone is a negative number and timezones west of Greenwich it is a positive number. 

```
foglamp@foglamp-18:~$ cat t.c
main()
{
	extern long timezone;
	tzset();
	printf("Timezone: %ld\n", timezone);
}
foglamp@foglamp-18:~$ export TZ=EET
foglamp@foglamp-18:~$ ./t
Timezone: -7200
foglamp@foglamp-18:~$ export TZ=CET
foglamp@foglamp-18:~$ ./t
Timezone: -3600
foglamp@foglamp-18:~$ export TZ=EST
foglamp@foglamp-18:~$ ./t
Timezone: 18000
foglamp@foglamp-18:~$ 
```

Comment (MarkRiddoch) [reply C/common/reading.cpp]: Given the nature of this issue it would be good if you could try it in all your time zones.

Comment (AmandeepArora) [reply C/common/reading.cpp]: Ok, got it. Thanks.

**Pull request merged**